src/Sentry.Unity/SentryMonoBehaviour.cs

```diff
@@ -1,14 +1,44 @@
 using System;
+using System.Collections;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
 using Sentry.Unity.Integrations;
 using UnityEngine;
 
 namespace Sentry.Unity
 {
+    /// <summary>
+    /// Singleton and DontDestroyOnLoad setup.
+    /// </summary>
+    [SuppressMessage("ReSharper", "ClassNeverInstantiated.Global")]
+    internal partial class SentryMonoBehaviour
+    {
+        private static SentryMonoBehaviour? Instance;
+
+        // ReSharper disable once UnusedMember.Local
+        private void Awake()
+        {
+            // Unity overrides `==` operator in MonoBehaviours
+            // ReSharper disable once ConvertIfStatementToNullCoalescingAssignment
+            if (Instance is null)
```

Comment:
I would use '==' instead of 'is' especially since Unity overrides the operator. It probably does not matter in this case but I'd use '==' for all things MonoBehaviour.